datahub-web-react/src/app/entity/shared/containers/profile/nav/ProfileNavBrowsePathV2.tsx

```diff
@@ -0,0 +1,109 @@
+            filters.push({ field: ORIGIN_FILTER_NAME, values: [entityData?.origin as FabricType] });
+        }
+        if (entityData?.platform) {
+            filters.push({ field: PLATFORM_FILTER_NAME, values: [entityData.platform.urn] });
+        }
+        return filters;
+    }
+
+    function generateFiltersForBrowsePath(path: string[]) {
+        const filters = generateFiltersForEnvironment();
+        const pathValue = createBrowseV2SearchFilter(path);
+        filters.push({ field: BROWSE_PATH_V2_FILTER_NAME, values: [pathValue] });
+        return filters;
+    }
+
+    return (
+        <BrowseRow>
+            <Breadcrumb style={{ fontSize: '16px' }} separator=">">
+                <BreadcrumbItem
+                    disabled={!isBrowsable}
+                    onClick={() => handlePathClick([{ field: ENTITY_SUB_TYPE_FILTER_NAME, values: [type] }])}
+                >
+                    {entityRegistry.getCollectionName(type)}
+                </BreadcrumbItem>
+                {hasMultipleEnvironments && hasEnvironment && (
+                    <BreadcrumbItem
+                        disabled={!isBrowsable}
+                        onClick={() =>
+                            handlePathClick([
+                                { field: ENTITY_SUB_TYPE_FILTER_NAME, values: [type] },
+                                { field: ORIGIN_FILTER_NAME, values: [entityData?.origin as FabricType] },
+                            ])
+                        }
+                    >
+                        {entityData?.origin}
+                    </BreadcrumbItem>
+                )}
+                {entityData?.platform && (
+                    <BreadcrumbItem
+                        disabled={!isBrowsable}
+                        onClick={() => handlePathClick(generateFiltersForEnvironment())}
+                    >
+                        {entityRegistry.getDisplayName(EntityType.DataPlatform, entityData.platform)}
+                    </BreadcrumbItem>
+                )}
+                {entityData?.browsePathV2?.path.map((pathEntry, index) => (
+                    <BreadcrumbItem
+                        key={pathEntry?.name}
+                        disabled={!isBrowsable}
+                        onClick={() =>
+                            handlePathClick(
+                                generateFiltersForBrowsePath([
+                                    ...(entityData.browsePathV2 as BrowsePathV2).path
+                                        .slice(0, index + 1)
+                                        .map((e) => e?.name || ''),
+                                ]),
+                            )
+                        }
+                        data-testid={`browse-path-${pathEntry?.name}`}
+                    >
+                        {pathEntry?.entity
+                            ? entityRegistry.getDisplayName(pathEntry.entity.type, pathEntry.entity)
+                            : pathEntry?.name}
```

Comment:
do we need pathEntry? with a question mark? 

Can't we guarantee that the object exists?

Reply:
oh good call - this was just a type issue from our main graphql file - we can indeeed guarantee this thing isn't null